tools/homebrew-start/lanraragi

```diff
@@ -1,22 +1,40 @@
 #!/bin/bash
 
-trap "kill 0" SIGINT
-
-# prefix/place where lrr is installed & modify redis config
+# prefix/place where lrr is installed
 workdir=$(dirname $0)/$(dirname $(readlink $0))/..
 
+# default locations
 if [[ ${OSTYPE} == "darwin"* ]]; then
   _LRR_DATA_DIRECTORY="${HOME}/Library/Application Support/LANraragi"
-  sed -i "" "s~<LRR_DATA_DIRECTORY>~${_LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
+  _LRR_LOG_DIRECTORY="${HOME}/Library/Logs/LANraragi"
 else
   _LRR_DATA_DIRECTORY="${HOME}/LANraragi"
-  sed -i"" "s~<LRR_DATA_DIRECTORY>~${_LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
+  _LRR_LOG_DIRECTORY="${HOME}/LANraragi/log"
+fi
+
+# initialize if not set
+if [[ ${LRR_DATA_DIRECTORY} == "" ]]; then
+  export LRR_DATA_DIRECTORY="${_LRR_DATA_DIRECTORY}"
+fi
+if [[ ${LRR_LOG_DIRECTORY} == "" ]]; then
+  export LRR_LOG_DIRECTORY="${_LRR_LOG_DIRECTORY}"
 fi
-echo "Data Directory is set to '${_LRR_DATA_DIRECTORY}'"
 
-# create path if it does not exist yet
-mkdir -p "${_LRR_DATA_DIRECTORY}/log"
+# change values in redis.conf
+perl -pi.bak -e "s~<LRR_DATA_DIRECTORY>~${LRR_DATA_DIRECTORY}~g" ${workdir}/redis.conf
```

Comment:
What is the `-pi.bak` argument for here? afaik you can just run -e and it'll work as-is.

Reply:
We can leave the `.bak` part out (that just creates a backup file).
However, `-p` and `-i` are required.

Reply:
Right, re-read on my perl cli arguments and you're correct -- I don't use those too often :(